junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/DefaultParallelExecutionConfiguration.java

```diff
@@ -20,14 +23,21 @@
 	private final int maxPoolSize;
 	private final int corePoolSize;
 	private final int keepAliveSeconds;
+	private final Predicate<? super ForkJoinPool> saturatePredicate;
 
 	DefaultParallelExecutionConfiguration(int parallelism, int minimumRunnable, int maxPoolSize, int corePoolSize,
 			int keepAliveSeconds) {
+		this(parallelism, minimumRunnable, maxPoolSize, corePoolSize, keepAliveSeconds, null);
+	}
+
+	DefaultParallelExecutionConfiguration(int parallelism, int minimumRunnable, int maxPoolSize, int corePoolSize,
+			int keepAliveSeconds, Predicate<? super ForkJoinPool> saturatePredicate) {
```

Comment:
I don't think we need to change this class at all.

Reply:
I added this to allow a custom configuration to use the DefaultParallelExecutionConfiguration with a non-null saturate , as done in the unit test. Otherwise it has to create a subclass.

Reply:
This class is package-private, i.e., not usable by custom config classes.

Reply:
Duh, you're right. In fact, I had noticed this when I tried to implement a custom strategy for Camel and I had to create a class with all interface methods.
I suppose the interface is not so large or the use case so common to merit having an extendable base class?

Reply:
So, would you suggest that I remove the changes in the DefaultParallelExecutionConfiguration class and modify the unit test accordingly? I think that's the only remaining open point of the suggested changes.

Reply:
I think that's what he meant, yes :+1: